app/src/main/java/com/guzelgimadieva/tasky/AppViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package com.guzelgimadieva.tasky
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.guzelgimadieva.tasky.core.data.remote.model.AccessTokenRequest
+import com.guzelgimadieva.tasky.core.network.TaskyServiceImpl
+import com.guzelgimadieva.tasky.core.network.onError
+import com.guzelgimadieva.tasky.core.network.onSuccess
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class AppViewModel @Inject constructor() : ViewModel() {
+    private val service = TaskyServiceImpl()
```

Comment:
I would not create such instances directly in the ViewModel, it defeats the whole purpose of using dependency injection. Your VM will always be forced that way to use this specific instance. If you were to write a test case for example that uses a test instance of the API service, there would be no way for the test to pass that. Instead it should look like this:
```
@HiltViewModel
class AppViewModel @Inject constructor(
    private val service: TaskyService // Let Hilt inject the TaskyServiceImpl here
): ViewModel()
```